sdk/include/psdk/driverspecs.h

```diff
@@ -35,6 +34,12 @@
 #define _IRQL_uses_cancel_
 #define __drv_setsIRQL(irql)
 
+#undef _IRQL_inout_
+#define _IRQL_inout_                               _IRQL_saves_ _IRQL_restores_
+#define _IRQL_limited_to_(level)                   ASSERT(KeGetCurrentIrql() <= level); _Analysis_assume_(KeGetCurrentIrql() <= level);
```

Comment:
```suggestion
#define _IRQL_limited_to_(level) do { \
    ASSERT(KeGetCurrentIrql() <= (level)); \
    _Analysis_assume_(KeGetCurrentIrql() <= (level)); \
} while (0)
```

Reply:
These are not code lines to put e.g. within functions, but are code analyzer annotations that prefix the function prototypes: as such you cannot use the do-while(0) trick.

Reply:
No, these go into a function. You obviously can't call KeGetCurrentIrql() from outside of a function. I prefer keeping it as it is though, as MS does the same and the do/while could confuse the analyzer.

Reply:
Ah, thanks for the correction :+1: